billing/lib/customer-billing-queue.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * @param {import('./api').CustomerBillingInstruction} instruction 
+ * @param {{
+ *   subscriptionStore: import('./api').SubscriptionStore
+ *   consumerStore: import('./api').ConsumerStore
+ *   spaceBillingQueue: import('./api').SpaceBillingQueue
+ * }} ctx
+ * @returns {Promise<import('@ucanto/interface').Result>}
+ */
+export const handleCustomerBillingInstruction = async (instruction, ctx) => {
+  console.log(`Processing customer billing instruction for: ${instruction.customer}`)
+  console.log(`Period: ${instruction.from.toISOString()} - ${instruction.to.toISOString()}`)
+
+  /** @type {string|undefined} */
+  let cursor
+  while (true) {
+    const subsList = await ctx.subscriptionStore.list({ customer: instruction.customer }, { cursor })
+    if (subsList.error) return subsList
+
+    // TODO: this is going to be inefficient for any client with many spaces
+    // and may eventually cause billing to fail.
+    for (const s of subsList.ok.results) {
```

Comment:
Could we just iterate over subscription store as opposed to first iterating over customers and then getting all the subscriptions that customer has ?